### DISCUSSION TITLE
Switches with postfix "none"

### DISCUSSION THREAD

**Discussion by requireyan** [make-all/tuya-local#2686] [General]
What is the reason for some switches to have "none" postfix?
For example: **smartplugv2_polled_power** is adding a single switch and it's entity id will be *your_chosen_name*_none

**Comment by make-all** [comment]
It is the inconsistent way HA handles translations for device classes.

**Comment by requireyan** [comment]
This has been fixed in 2024.12.2, thank you.

**Reply by make-all** [reply]
Yes, I found an unadvertised function that could be used to query whether the entity would support class based name translations or not. Strangely the [translation file for switch](https://github.com/home-assistant/core/blob/dev/homeassistant/components/switch/strings.json) has names for the classes, even though the switch entity does not support them.
